packages/lesswrong/server/resolvers/userResolvers.ts

```diff
@@ -21,6 +21,82 @@ import { getAnalyticsConnection } from "../analytics/postgresConnection";
 import GraphQLJSON from 'graphql-type-json';
 import { rateLimitDateWhenUserNextAbleToComment, rateLimitDateWhenUserNextAbleToPost } from '../rateLimits/utils';
 import { RateLimitInfo } from '../rateLimits/types';
+import VotesRepo, { RecentVoteInfo } from '../repos/VotesRepo';
+import uniq from 'lodash/uniq';
+// import { groupBy } from 'underscore';
+
+export type RecentKarmaInfo = {
+  recentKarma: number, 
+  recentPostKarma: number,
+  recentCommentKarma: number,
+  downvoterCount: number, 
+  postDownvoterCount: number,
+  commentDownvoterCount: number,
+}
+
+async function getLatest20contentItems (votes: RecentVoteInfo[]): Promise<RecentVoteInfo[]> {
```

Comment:
Are we doing this because getting 10 comments & 10 posts doesn't guarantee 20 items, if the user has less than 10 of one of them?

Reply:
If you don't do this filter, then a user who's ever made one good post will forever have that post be counted as part of their last 20 items, which makes the filter much less useful to detect people who've recently been bad.